src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -0,0 +1,193 @@
+        repositoryService.deleteAllContentInRepository(templateRepo);
+        repositoryService.deleteAllContentInRepository(solutionRepo);
+        repositoryService.deleteAllContentInRepository(testRepo);
+        copyExerciseContentToRepository(templateRepo, RepositoryType.TEMPLATE, basePath);
+        copyExerciseContentToRepository(solutionRepo, RepositoryType.SOLUTION, basePath);
+        copyExerciseContentToRepository(testRepo, RepositoryType.TESTS, basePath);
+    }
+
+    /**
+     * copies everything from the extracted zip file to the repository, except the .git folder
+     *
+     * @param repository     the repository to which the content should be copied
+     * @param repositoryType the type of the repository
+     * @param basePath       the path to the extracted zip file
+     **/
+    private void copyExerciseContentToRepository(Repository repository, RepositoryType repositoryType, Path basePath) throws IOException {
+        FileUtils.copyDirectory(retrieveRepositoryDirectoryPath(basePath, repositoryType.getName()).toFile(), repository.getLocalPath().toFile(),
+                new NotFileFilter(new NameFileFilter(".git")));
+        repository.setContent(null);
+    }
+
+    private ProgrammingExercise getProgrammingExerciseFromDetailsFile(Path extractedZipPath) throws IOException {
+        var exerciseJsonPath = retrieveExerciseJsonPath(extractedZipPath);
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        objectMapper.findAndRegisterModules();
+
+        try {
+            return objectMapper.readValue(exerciseJsonPath.toFile(), ProgrammingExercise.class);
+        }
+        catch (IOException e) {
+            throw new BadRequestAlertException("The JSON file for the programming exercise is not valid or was not found.", "programmingExercise", "exerciseJsonNotValidOrFound");
+        }
+    }
+
+    private void checkRepositoriesExist(Path path) throws IOException {
+        List<String> result;
+        try (Stream<Path> stream = Files.walk(path)) {
+            result = stream.filter(Files::isDirectory).map(f -> f.getFileName().toString())
+                    .filter(name -> name.endsWith("-exercise") || name.endsWith("-tests") || name.endsWith("-solution")).toList();
+        }
+
+        if (result.size() != 3) {
+            throw new BadRequestAlertException("The zip file doesn't contain the template, solution or tests repository or they do not follow the naming scheme.",
+                    "programmingExercise", "repositoriesInZipNotValid");
+        }
+
+    }
+
+    private Path retrieveRepositoryDirectoryPath(Path dirPath, String repoType) {
+        List<Path> result;
+        try (Stream<Path> walk = Files.walk(dirPath)) {
+            result = walk.filter(Files::isDirectory).filter(f -> f.getFileName().toString().endsWith("-" + repoType)).filter(f -> !f.getFileName().endsWith(".zip")).toList();
```

Comment:
The `endsWith(.zip)` filter is redundant since you already check for `endsWith(-repoType)` before that.